File: app/src/main/java/pl/adriankremski/coolector/TheApp.java
package pl.adriankremski.coolector;

import android.app.Application;
import android.content.Context;

import pl.adriankremski.coolector.dagger.AppComponent;
import pl.adriankremski.coolector.dagger.AppModule;
import pl.adriankremski.coolector.dagger.DaggerAppComponent;

public class TheApp extends Application{

    private AppComponent appComponent;

    public static TheApp get(Context context) {
        return (TheApp) context.getApplicationContext();
    }

    @Override
    public void onCreate() {
        super.onCreate();
        initAppComponent();
    }

    private void initAppComponent() {
        appComponent = DaggerAppComponent.builder()
                .appModule(new AppModule(this))
                .build();
    }


    public AppComponent getAppComponent() {
        return appComponent;
    }
}

File: app/src/main/java/pl/adriankremski/coolector/TheApp.kt
package pl.adriankremski.coolector

import android.content.Context
import android.support.multidex.MultiDexApplication
import pl.adriankremski.coolector.dagger.AppComponent
import pl.adriankremski.coolector.dagger.AppModule
import pl.adriankremski.coolector.dagger.DaggerAppComponent

class TheApp : MultiDexApplication() {

    companion object {
        operator fun get(context: Context): TheApp {
            return context.applicationContext as TheApp
        }
    }

    var appComponent: AppComponent? = null

    override fun onCreate() {
        super.onCreate()
        initAppComponent()
    }

    private fun initAppComponent() {
        appComponent = DaggerAppComponent.builder()
                .appModule(AppModule(this))
                .build()
    }
}
